### PR TITLE
API-1437 End user cannot see trusted applications on their list of authorised applications

### PR DESCRIPTION
**Pull request by romankusnierz**

### CODE REVIEW

File: test/acceptance/NavigationSugar.scala [hunk 1]
```diff
@@ -46,6 +46,10 @@ trait NavigationSugar extends WebBrowser with Eventually with Assertions with Ma
     webDriver.findElement(locator).click()
   }
 
+  def verifySize(selectorId: String, expectedSize: Int)(implicit webDriver: WebDriver) = {
+    webDriver.findElements(By.cssSelector(s"[$selectorId] > li")).size() shouldBe expectedSize
+  }
```
Comment (hughfdjackson) [inline test/acceptance/NavigationSugar.scala]: Perhaps `verifyListSize` would be more descriptive?  `verifySize` of a selector could be the characters, number of children (p, li, etc).

File: app/controllers/Revocation.scala [hunk 2]
```diff
@@ -41,19 +40,18 @@ trait Revocation extends FrontendController with Authentication {
   }
 
   val listAuthorizedApplications = authenticated.async { implicit user => implicit request =>
-    delegatedAuthorityConnector.fetchApplicationAuthorities()
-      .map(applications => Ok(views.html.revocation.authorizedApplications(applications)))
+    revocationService.fetchUntrustedApplicationAuthorities()
+      .map(applications => Ok(views.html.revocation.authorizedApplications(applications.filter(!_.application.trusted))))
```
Comment (tom59) [inline app/controllers/Revocation.scala]: filter twice

File: test/acceptance/specs/RevokeApplicationAuthoritySpec.scala [hunk 3]
```diff
@@ -80,9 +75,27 @@ class RevokeApplicationAuthoritySpec extends BaseSpec with NavigationSugar {
...
+      LoginStub.stubSuccessfulLogin()
+      DelegatedAuthorityStub.stubSuccessfulFetchApplicationAuthorities(Seq(trustedApp))
+      DelegatedAuthorityStub.stubSuccessfulFetchApplicationAuthority(trustedApp)
+
+      go(AuthorizedApplicationsPage)
+
+      go(WithdrawPermissionPage(trustedApp.application.id))
+
+      on(TechnicalDifficultiesPage)
+    }
```
Comment (hughfdjackson) [inline test/acceptance/specs/RevokeApplicationAuthoritySpec.scala]: Is this a different behaviour than an attempt to revoke an app that doesn't exist?

File: test/unit/connectors/DelegatedAuthorityConnectorSpec.scala [hunk 4]
```diff
@@ -91,28 +90,29 @@ class DelegatedAuthorityConnectorSpec extends UnitSpec with Matchers with ScalaF
...
-             |      "name":"$scopeName",
-             |      "description":"$scopeDescription"
+             |      "key":"${scope.key}",
+             |      "name":"${scope.name}",
+             |      "description":"${scope.description}"
              |    }
              |  ],
              |  "earliestGrantDate":$earliestGrantDate
               |}""".stripMargin)
       ))
```
Comment (hughfdjackson) [inline test/unit/connectors/DelegatedAuthorityConnectorSpec.scala]: Is it worth factoring out these stubs into a private method?

File: test/acceptance/specs/RevokeApplicationAuthoritySpec.scala (hunk 3 above)
Comment (romankusnierz) [inline test/acceptance/specs/RevokeApplicationAuthoritySpec.scala]: It should be the same from the end user's perspective and it is as far as I'm aware

File: test/acceptance/NavigationSugar.scala (hunk 1 above)
Comment (romankusnierz) [inline test/acceptance/NavigationSugar.scala]: That's true, I'll make it more descriptive

File: test/unit/connectors/DelegatedAuthorityConnectorSpec.scala (hunk 4 above)
Comment (romankusnierz) [inline test/unit/connectors/DelegatedAuthorityConnectorSpec.scala]: That is a good candidate for a quick chat within the team I guess as you could even go further and reuse the stubs from component tests.

File: app/controllers/Revocation.scala (hunk 2 above)
Comment (romankusnierz) [inline app/controllers/Revocation.scala]: Good spot!

**Pull request merged**